components/style/values/specified/counters.rs

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Specified types for counter properties.
+
+use cssparser::Parser;
+use parser::{Parse, ParserContext};
+use std::fmt;
+use style_traits::{ParseError, ToCss};
+use values::CustomIdent;
+use values::computed::{Context, CounterReset as ComputedValue, ToComputedValue};
+use values::specified::Integer;
+
+/// A specified value for the `counter-reset` property.
+#[cfg_attr(feature = "gecko", derive(MallocSizeOf))]
+#[derive(Clone, Debug, PartialEq)]
+pub struct CounterReset(pub Vec<(CustomIdent, super::Integer)>);
```

Comment:
nit: No need to use `super::Integer`, just use `Integer`, since you're importing it too.